src/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.cs

```diff
@@ -1202,18 +1202,9 @@ internal Object InternalGetValue(String name, Object defaultValue, bool doNotExp
                         // make sure the string is null terminated before processing the data
                         if (blob.Length > 0 && blob[blob.Length - 1] != (char)0)
                         {
-                            try
-                            {
-                                char[] newBlob = new char[checked(blob.Length + 1)];
-                                Array.Copy(blob, 0, newBlob, 0, blob.Length);
-                                newBlob[newBlob.Length - 1] = (char)0;
-                                blob = newBlob;
-                            }
-                            catch (OverflowException e)
-                            {
-                                throw new IOException(SR.Arg_RegGetOverflowBug, e);
-                            }
-                            blob[blob.Length - 1] = (char)0;
+                            char[] newBlob = new char[blob.Length + 1];
+                            Array.Copy(blob, 0, newBlob, 0, blob.Length);
+                            blob = newBlob;
```

Comment:
According to the latest code coverage report, this if block is not exercised by any tests.  Can you add a test that exercises it?  While the change looks safe, it'd be good to have some test coverage to verify that.

Also, could this be simplified further to just be:
```C#
Array.Resize(ref blob, blob.Length + 1);
```
?